libqaul/src/auth/mod.rs

```diff
@@ -1 +1,12 @@
+pub struct Authenticator;
+use crate::{
+    error::QaulResult,
+    users::User,
+    Qaul,
+};
 
+impl Qaul {
+    pub fn authenticate(&self, token: &str) -> QaulResult<Option<User>> {
```

Comment:
Have a look at the API that I added in #247 (i.e. maybe rebase `master` :stuck_out_tongue:). There's `user_login` and `user_logout`, plus the concept of an `AuthUser` (i.e. something that wraps both the `Identity` and `Token` of a session).

Something that function currently lacks (which...feel free to add it!) is a `secret` argument, i.e. a password.

Reply:
Hmmm that won't quite work in this case as we don't have the identity of the user we're logging in, just an api key

with this when you start an application you do normal first party login to get the token, which is certainly doable, but also seems somewhat of a pain (and probably encourages applications to store the secret to prevent the user from having to log in every time)

Reply:
or is the idea that that part is stored in the api layer?
i know there's some concept of logging in/logging out for the crypto things y'all are doing so it'd be a world of several keys regardless